tests/yam/agama/boot_encrypted.pm

```diff
@@ -0,0 +1,44 @@
+## Copyright 2024 SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+# Summary: Boot the system for encryption scenario.
+# - {Enter passphrase for root partition for full disk encryption}.
+# - Select first entry to boot in grub2
+# - {Enter passphrase for swap partition for full disk encryption}.
+# = {Enter passphrase for home partition for home encryption}.
+# Maintainer: QE YaST and Migration (QE Yam) <qe-yam at suse de>
+
+use base 'opensusebasetest';
+use strict;
+use warnings;
+use testapi;
+
+sub run {
+    my $enter_passphrase_for_root = $testapi::distri->get_enter_passphrase_for_root();
+    my $enter_passphrase_for_swap = $testapi::distri->get_enter_passphrase_for_swap();
+    my $enter_passphrase_for_home = $testapi::distri->get_enter_passphrase_for_home();
+    my $grub_menu = $testapi::distri->get_grub_menu_installed_system();
+
+    if (get_var('ENCRYPTED_PARTITIONS_CONTAIN_ROOT')) {
```

Comment:
you didn't get fully the idea, having `ENCRYPTED_PARTITIONS` and doing a regex with the value of the setting, we can avoid having 2 additional settings.